### PR TITLE
Add loop to read all data in rcv buffer

### PR DESCRIPTION
**Pull request by chrysh**
Previously, read would be called only once. This was a problem for devices that had short packages. This commit makes sure the read is continued until the read buffer is empty.

### CODE REVIEW

File: linux-serial-test.c [hunk 1]
```diff
@@ -532,35 +532,42 @@ static unsigned char next_count_value(unsigned char c)
...
+                                                dump_serial_port_stats();
+                                                exit(-EIO);
+                                        }
+                                        _read_count_value = rb[i];
+                                }
+                                _read_count_value = next_count_value(_read_count_value);
+                        }
+                        _read_count += c;
+                        actual_read_count += c;
+                } else {
```
Comment (andy-shev) [inline linux-serial-test.c:566]: `read()` can be interrupted by a signal. Here you break the loop even when it's not needed (EAGAIN IIRC and other error codes should be handled).

Comment (andy-shev) [reply linux-serial-test.c]: Looks like that, thank you!
P.S> I haven't realized it's you :-) Hello there, interesting clash in the open source world.

Comment (chrysh) [reply linux-serial-test.c]: I did not know you were doing basically everything AND linux-serial-test, how do you even find the time?

**Comment by cbrake** [comment]
thanks for the contribution @chrysh  and the review @andy-shev !

**Pull request merged**